// Generated by the protocol buffer compiler.  DO NOT EDIT!
// source: google/genomics/v1/variants.proto

package com.google.genomics.v1;

public interface ImportVariantsRequestOrBuilder extends
    // @@protoc_insertion_point(interface_extends:google.genomics.v1.ImportVariantsRequest)
    com.google.protobuf.MessageOrBuilder {

  /**
   * <code>optional string variant_set_id = 1;</code>
   *
   * <pre>
   * Required. The variant set to which variant data should be imported.
   * </pre>
   */
  java.lang.String getVariantSetId();
  /**
   * <code>optional string variant_set_id = 1;</code>
   *
   * <pre>
   * Required. The variant set to which variant data should be imported.
   * </pre>
   */
  com.google.protobuf.ByteString
      getVariantSetIdBytes();

  /**
   * <code>repeated string source_uris = 2;</code>
   *
   * <pre>
   * A list of URIs referencing variant files in Google Cloud Storage. URIs can
   * include wildcards [as described here](https://cloud.google.com/storage/docs/gsutil/addlhelp/WildcardNames).
   * Note that recursive wildcards ('**') are not supported.
   * </pre>
   */
  com.google.protobuf.ProtocolStringList
      getSourceUrisList();
  /**
   * <code>repeated string source_uris = 2;</code>
   *
   * <pre>
   * A list of URIs referencing variant files in Google Cloud Storage. URIs can
   * include wildcards [as described here](https://cloud.google.com/storage/docs/gsutil/addlhelp/WildcardNames).
   * Note that recursive wildcards ('**') are not supported.
   * </pre>
   */
  int getSourceUrisCount();
  /**
   * <code>repeated string source_uris = 2;</code>
   *
   * <pre>
   * A list of URIs referencing variant files in Google Cloud Storage. URIs can
   * include wildcards [as described here](https://cloud.google.com/storage/docs/gsutil/addlhelp/WildcardNames).
   * Note that recursive wildcards ('**') are not supported.
   * </pre>
   */
  java.lang.String getSourceUris(int index);
  /**
   * <code>repeated string source_uris = 2;</code>
   *
   * <pre>
   * A list of URIs referencing variant files in Google Cloud Storage. URIs can
   * include wildcards [as described here](https://cloud.google.com/storage/docs/gsutil/addlhelp/WildcardNames).
   * Note that recursive wildcards ('**') are not supported.
   * </pre>
   */
  com.google.protobuf.ByteString
      getSourceUrisBytes(int index);

  /**
   * <code>optional .google.genomics.v1.ImportVariantsRequest.Format format = 3;</code>
   *
   * <pre>
   * The format of the variant data being imported. If unspecified, defaults to
   * to `VCF`.
   * </pre>
   */
  int getFormatValue();
  /**
   * <code>optional .google.genomics.v1.ImportVariantsRequest.Format format = 3;</code>
   *
   * <pre>
   * The format of the variant data being imported. If unspecified, defaults to
   * to `VCF`.
   * </pre>
   */
  com.google.genomics.v1.ImportVariantsRequest.Format getFormat();

  /**
   * <code>optional bool normalize_reference_names = 5;</code>
   *
   * <pre>
   * Convert reference names to the canonical representation.
   * hg19 haploytypes (those reference names containing "_hap")
   * are not modified in any way.
   * All other reference names are modified according to the following rules:
   * The reference name is capitalized.
   * The "chr" prefix is dropped for all autosomes and sex chromsomes.
   * For example "chr17" becomes "17" and "chrX" becomes "X".
   * All mitochondrial chromosomes ("chrM", "chrMT", etc) become "MT".
   * </pre>
   */
  boolean getNormalizeReferenceNames();
}
