// Generated by the protocol buffer compiler.  DO NOT EDIT!
// source: google/genomics/v1/variants.proto

package com.google.genomics.v1;

public interface VariantOrBuilder extends
    // @@protoc_insertion_point(interface_extends:google.genomics.v1.Variant)
    com.google.protobuf.MessageOrBuilder {

  /**
   * <code>optional string variant_set_id = 15;</code>
   *
   * <pre>
   * The ID of the variant set this variant belongs to.
   * </pre>
   */
  java.lang.String getVariantSetId();
  /**
   * <code>optional string variant_set_id = 15;</code>
   *
   * <pre>
   * The ID of the variant set this variant belongs to.
   * </pre>
   */
  com.google.protobuf.ByteString
      getVariantSetIdBytes();

  /**
   * <code>optional string id = 2;</code>
   *
   * <pre>
   * The Google generated ID of the variant, immutable.
   * </pre>
   */
  java.lang.String getId();
  /**
   * <code>optional string id = 2;</code>
   *
   * <pre>
   * The Google generated ID of the variant, immutable.
   * </pre>
   */
  com.google.protobuf.ByteString
      getIdBytes();

  /**
   * <code>repeated string names = 3;</code>
   *
   * <pre>
   * Names for the variant, for example a RefSNP ID.
   * </pre>
   */
  com.google.protobuf.ProtocolStringList
      getNamesList();
  /**
   * <code>repeated string names = 3;</code>
   *
   * <pre>
   * Names for the variant, for example a RefSNP ID.
   * </pre>
   */
  int getNamesCount();
  /**
   * <code>repeated string names = 3;</code>
   *
   * <pre>
   * Names for the variant, for example a RefSNP ID.
   * </pre>
   */
  java.lang.String getNames(int index);
  /**
   * <code>repeated string names = 3;</code>
   *
   * <pre>
   * Names for the variant, for example a RefSNP ID.
   * </pre>
   */
  com.google.protobuf.ByteString
      getNamesBytes(int index);

  /**
   * <code>optional int64 created = 12;</code>
   *
   * <pre>
   * The date this variant was created, in milliseconds from the epoch.
   * </pre>
   */
  long getCreated();

  /**
   * <code>optional string reference_name = 14;</code>
   *
   * <pre>
   * The reference on which this variant occurs.
   * (such as `chr20` or `X`)
   * </pre>
   */
  java.lang.String getReferenceName();
  /**
   * <code>optional string reference_name = 14;</code>
   *
   * <pre>
   * The reference on which this variant occurs.
   * (such as `chr20` or `X`)
   * </pre>
   */
  com.google.protobuf.ByteString
      getReferenceNameBytes();

  /**
   * <code>optional int64 start = 16;</code>
   *
   * <pre>
   * The position at which this variant occurs (0-based).
   * This corresponds to the first base of the string of reference bases.
   * </pre>
   */
  long getStart();

  /**
   * <code>optional int64 end = 13;</code>
   *
   * <pre>
   * The end position (0-based) of this variant. This corresponds to the first
   * base after the last base in the reference allele. So, the length of
   * the reference allele is (end - start). This is useful for variants
   * that don't explicitly give alternate bases, for example large deletions.
   * </pre>
   */
  long getEnd();

  /**
   * <code>optional string reference_bases = 6;</code>
   *
   * <pre>
   * The reference bases for this variant. They start at the given
   * position.
   * </pre>
   */
  java.lang.String getReferenceBases();
  /**
   * <code>optional string reference_bases = 6;</code>
   *
   * <pre>
   * The reference bases for this variant. They start at the given
   * position.
   * </pre>
   */
  com.google.protobuf.ByteString
      getReferenceBasesBytes();

  /**
   * <code>repeated string alternate_bases = 7;</code>
   *
   * <pre>
   * The bases that appear instead of the reference bases.
   * </pre>
   */
  com.google.protobuf.ProtocolStringList
      getAlternateBasesList();
  /**
   * <code>repeated string alternate_bases = 7;</code>
   *
   * <pre>
   * The bases that appear instead of the reference bases.
   * </pre>
   */
  int getAlternateBasesCount();
  /**
   * <code>repeated string alternate_bases = 7;</code>
   *
   * <pre>
   * The bases that appear instead of the reference bases.
   * </pre>
   */
  java.lang.String getAlternateBases(int index);
  /**
   * <code>repeated string alternate_bases = 7;</code>
   *
   * <pre>
   * The bases that appear instead of the reference bases.
   * </pre>
   */
  com.google.protobuf.ByteString
      getAlternateBasesBytes(int index);

  /**
   * <code>optional double quality = 8;</code>
   *
   * <pre>
   * A measure of how likely this variant is to be real.
   * A higher value is better.
   * </pre>
   */
  double getQuality();

  /**
   * <code>repeated string filter = 9;</code>
   *
   * <pre>
   * A list of filters (normally quality filters) this variant has failed.
   * `PASS` indicates this variant has passed all filters.
   * </pre>
   */
  com.google.protobuf.ProtocolStringList
      getFilterList();
  /**
   * <code>repeated string filter = 9;</code>
   *
   * <pre>
   * A list of filters (normally quality filters) this variant has failed.
   * `PASS` indicates this variant has passed all filters.
   * </pre>
   */
  int getFilterCount();
  /**
   * <code>repeated string filter = 9;</code>
   *
   * <pre>
   * A list of filters (normally quality filters) this variant has failed.
   * `PASS` indicates this variant has passed all filters.
   * </pre>
   */
  java.lang.String getFilter(int index);
  /**
   * <code>repeated string filter = 9;</code>
   *
   * <pre>
   * A list of filters (normally quality filters) this variant has failed.
   * `PASS` indicates this variant has passed all filters.
   * </pre>
   */
  com.google.protobuf.ByteString
      getFilterBytes(int index);

  /**
   * <code>map&lt;string, .google.protobuf.ListValue&gt; info = 10;</code>
   *
   * <pre>
   * A map of additional variant information. This must be of the form
   * map&lt;string, string[]&gt; (string key mapping to a list of string values).
   * </pre>
   */
  java.util.Map<java.lang.String, com.google.protobuf.ListValue>
  getInfo();

  /**
   * <code>repeated .google.genomics.v1.VariantCall calls = 11;</code>
   *
   * <pre>
   * The variant calls for this particular variant. Each one represents the
   * determination of genotype with respect to this variant.
   * </pre>
   */
  java.util.List<com.google.genomics.v1.VariantCall> 
      getCallsList();
  /**
   * <code>repeated .google.genomics.v1.VariantCall calls = 11;</code>
   *
   * <pre>
   * The variant calls for this particular variant. Each one represents the
   * determination of genotype with respect to this variant.
   * </pre>
   */
  com.google.genomics.v1.VariantCall getCalls(int index);
  /**
   * <code>repeated .google.genomics.v1.VariantCall calls = 11;</code>
   *
   * <pre>
   * The variant calls for this particular variant. Each one represents the
   * determination of genotype with respect to this variant.
   * </pre>
   */
  int getCallsCount();
  /**
   * <code>repeated .google.genomics.v1.VariantCall calls = 11;</code>
   *
   * <pre>
   * The variant calls for this particular variant. Each one represents the
   * determination of genotype with respect to this variant.
   * </pre>
   */
  java.util.List<? extends com.google.genomics.v1.VariantCallOrBuilder> 
      getCallsOrBuilderList();
  /**
   * <code>repeated .google.genomics.v1.VariantCall calls = 11;</code>
   *
   * <pre>
   * The variant calls for this particular variant. Each one represents the
   * determination of genotype with respect to this variant.
   * </pre>
   */
  com.google.genomics.v1.VariantCallOrBuilder getCallsOrBuilder(
      int index);
}
